models/org_team.go

```diff
@@ -194,9 +194,25 @@ func (t *Team) RemoveRepository(repoID int64) error {
 	return sess.Commit()
 }
 
+func IsUsableTeamName(name string) (err error) {
+	var reservedTeamNames = []string{"new"}
+
+	for i := range reservedTeamNames {
+		if name == reservedTeamNames[i] {
+			return ErrNameReserved{name}
+		}
+	}
+
+	return nil
+}
+
 // NewTeam creates a record of new team.
 // It's caller's responsibility to assign organization ID.
-func NewTeam(t *Team) error {
+func NewTeam(t *Team) (err error) {
+	if err = IsUsableTeamName(t.Name); err != nil {
```

Comment:
My suggestion was actually to move the `len(t.Name) == 0` check inside `IsUsableTeamName` (as empty is clearly not a usable team name)